### PR TITLE
MGDOBR-310: add retry to rhoas client

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-310](https://issues.redhat.com/browse/MGDOBR-310) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java [hunk 1]
```diff
@@ -34,8 +44,9 @@ public Topic createTopicAndGrantAccessFor(String topicName, RhoasTopicAccessType
             NewTopicInput newTopicInput = new NewTopicInput()
                     .name(topicName)
                     .settings(new TopicSettings().numPartitions(1));
-
-            return rhoasClient.createTopicAndGrantAccess(newTopicInput, rhoasOpsAccountClientId, accessType).await().atMost(Duration.ofSeconds(rhoasTimeout));
+            return rhoasClient.createTopicAndGrantAccess(newTopicInput, rhoasOpsAccountClientId, accessType)
```
Comment (robpblake) [inline manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java:47]: @r00ta What happens if the Topic creation has succeeded in the first attempt, but the ACL creation fails? Is the creation of the Topic idempotent e.g. the retry doesn't get an error if the topic already exists?

Comment (r00ta) [reply manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java:47]: @robpblake If the topic has been created but the ACL creation fails the `createTopicAndGrantAccess` will revert the changes deleting the topic. 

Anyway, the rhoas client api are idempotent in general

**Review by lucamolteni — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java (hunk 1 above)
Comment (robpblake) [reply manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java:47]: Thanks @r00ta

**Review by robpblake — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java [hunk 2]
```diff
@@ -22,6 +22,16 @@
 
     @ConfigProperty(name = "rhoas.timeout-seconds")
     int rhoasTimeout;
+
+    @ConfigProperty(name = "rhoas.max_retries")
```
Comment (kostola) [inline manager/src/main/java/com/redhat/service/bridge/manager/RhoasServiceImpl.java:26]: I know it's too late but to be consistent with property naming so far this should be `rhoas.max-retries`

File: manager/src/main/resources/application.properties [hunk 3]
```diff
@@ -97,5 +97,9 @@ managed-connectors.auth.offline-token=${MC_AUTH_OFFLINE_TOKEN:fakeOfflineToken}
 rhoas.timeout-seconds=60
 rhoas.ops-account.client-id=fakeClientId
 
+rhoas.max_retries=5
```
Comment (kostola) [inline manager/src/main/resources/application.properties:100]: same here

**Pull request merged**